Add docs about mirroring channels (#12583)
conda/conda: docs/source/user-guide/configuration/mirroring.rst

==================
Mirroring channels
==================

The conda configuration system has several keys that can be used to set up a mirrored context.

The default setup
=================

By default, ``conda`` can serve packages from two main locations:

- ``repo.anaconda.com``: this is where ``defaults`` points to by default. This base location is hardcoded in the default value of ``default_channels``:
    - ``https://repo.anaconda.com/pkgs/main``
    - ``https://repo.anaconda.com/pkgs/r``
    - ``https://repo.anaconda.com/pkgs/msys2``

> - ``conda.anaconda.org``: this is where community channels like ``conda-forge`` or ``bioconda`` are looked up. This base location can be configured via ``channel_alias``.

Reviewer: Reading over the whole page, "hosted" isn't right. I'm okay with "looked up" in general, but still suggest "accessed", perhaps. Searched?

Or, maybe "this is where community channels like `conda-forge` or `bioconda` are hosted and looked up (searched?) by conda". Just thinking out loud.

> - ``conda.anaconda.org``: this is where community channels like ``conda-forge`` or ``bioconda`` are looked up. This base location can be configured via ``channel_alias``.

Author: The idea is that `conda.anaconda.org` is the default _host_ where these channel names expected to be served:

- `conda-forge` -> `https://conda.anaconda.org/conda-forge`
- `bioconda` -> `https://conda.anaconda.org/bioconda`

The user can change this default server name to something else (e.g. `my-mirror.org`) and then it would look like:

- `conda-forge` -> `https://my-mirror.org/conda-forge`
- `bioconda` -> `https://my-mirror.org/bioconda`

I still think that "look up" would be an appropriate term here, given how common it is in DNS contexts (e.g. ["Reverse DNS lookup"](https://en.wikipedia.org/wiki/Reverse_DNS_lookup). It's not a search either, where we try different options, but a single request that will either succeed or fail. Is there a better term for that?

> - ``conda.anaconda.org``: this is where community channels like ``conda-forge`` or ``bioconda`` are looked up. This base location can be configured via ``channel_alias``.

Reviewer: Yeah, "look up" is probably a good term to use here. I just wasn't super familiar with it, so it tripped me up. 

What about "this is where conda looks up community channels like `conda-forge` or `bioconda`."? That would also remove the passive voice, which is a plus.

> - ``conda.anaconda.org``: this is where community channels like ``conda-forge`` or ``bioconda`` are looked up. This base location can be configured via ``channel_alias``.

Author: In that case, we might want to say `conda` clients (to include `mamba`, `micromamba`, `rattler`, etc), which might add unnecessary noise? I'll leave it you, happy either way.

> - ``conda.anaconda.org``: this is where community channels like ``conda-forge`` or ``bioconda`` are looked up. This base location can be configured via ``channel_alias``.

Reviewer: Good point. I don't think that adds extra noise. To be clear, then it would be "this is where conda clients look up community channels..."? That works for me.

> - ``conda.anaconda.org``: this is where community channels like ``conda-forge`` or ``bioconda`` are looked up. This base location can be configured via ``channel_alias``.

Author: ```suggestion
- ``conda.anaconda.org``: this is where conda clients look up community channels like ``conda-forge`` or ``bioconda``. This base location can be configured via ``channel_alias``.
```

> So, when it comes to mirroring these channels, you have two account for those two locations.

Author: ```suggestion
So, when it comes to mirroring these channels, you have to account for those two locations.
```

Yea, oops :D

Mirror ``defaults``
===================

> Use ``default_channels`` to overwrite the default configuration. For example:

Author: ```suggestion
Use ``default_channels`` to overwrite the :doc:`default configuration <configuration>`_. For example:
```
